ulx/src/main/java/com/uplink/ulx/bridge/network/controller/NetworkController.java

```diff
@@ -1331,6 +1377,9 @@ private void scheduleInternetUpdatePacket(
     ) {
         final InternetUpdatePacket packet = createInternetUpdatePacket(internetHopCount);
 
+        final boolean isInternetReachable = internetHopCount != RoutingTable.HOP_COUNT_INFINITY;
+        notifyOnInternetConnectionChanged(isInternetReachable);
```

Comment:
This seems to be the correct moment for notifying about internet connection change, but I would'n do it inside the method called `scheduleInternetUpdatePacket()`, since now there is another important thing that it is doing. Maybe it would be nice to rename the method or to refactor the code in other way, so that no one gets confused later.

Reply:
yea, I was trying to find some place to do it that was already being invoked on every state change.  should it be called alongside every `scheduleInternetUpdatePacket()`?